### ISSUE TITLE
Titles Without inetrefs Display In All Recordings, But Not Individually

### ISSUE THREAD

**Issue by billmeek** [MythTV-Clients/MythtvPlayerForAndroid#26]
In this test case, the title 'Noticiero Univisión' should show up
between 'NCIS: New Orleans' and 'PBS NewsHour Weekend'.

[This was actually a test of special characters displaying
 correctly (and they do if the backend has $LANG set
 properly.)]

```
04-01 18:40:47.596  20394-20449/org.mythtv.android V/DvrPersistenceServiceEventHandler﹕ updateRecordedPrograms : program=Program{startTime=2015-04-01T22:30:00.000Z, endTime=2015-04-01T23:00:00.000Z, title='Noticiero Univisión', subTitle='', category='News', catType='series', repeat=true, videoProps=1, audioProps=0, subProps=1, seriesId='', programId='SH000193110000', stars=0.0, fileSize=671304008, lastModified=2015-04-01T23:02:59.000Z, programFlags=4101, fileName='2801_20150401223000.ts', hostName='mc0', airdate=1998-01-01, description='Noticias del Mundo de última hora, comentarios y opinión, deportes, política, trabajos y educación.', inetref='', season=0, episode=0, totalEpisodes=null, channel=ChannelInfo{chanId=2801, chanNum='801', callSign='WGBO', iconURL='', channelName='WGBODT (WGBO-DT)', mplexId=32767, serviceId=0, aTSCMajorChan=801, aTSCMinorChan=0, format='Default', frequencyId='801', fineTune=0, chanFilters='', sourceId=2, inputId=0, commFree=false, useEIT=false, visible=true, xMLTVID='42635', defaultAuth='', programs=[]}, recording=RecordingInfo{recordedId=2765, status=-3, priority=0, startTs=2015-04-01T22:30:00.000Z, endTs=2015-04-01T23:00:00.000Z, recordId=1587, recGroup='Default', playGroup='Default', storageGroup='Default', recType=0, dupInType=15, dupMethod=6, encoderId=0, encoderName='', profile='Default'}, artworkInfos=[], castMembers=[CastMember{name='Jorge Ramos', characterName='', role='', translatedRole=''}, CastMember{name='María Elena Salinas', characterName='', role='', translatedRole=''}]}
```

**Comment by dmfrey** [comment]
did it not show up at all? or just out of order?

Inetref is the linkage between the title list and recordings.  How would the title list endpoint not return an inetref?  They only return those 2 fields.

**Comment by billmeek** [comment]
They don't show up at, except in the All Recordings list.

Some titles have no inetref, e.g. news programs. Shows currently recording
won't have one 'til mythmetadata runs. Specials frequently don't have them.

There's a group of users on the -mythtv mailing list that turn off mythmetadatalookup.

Examples:
``` sql
SELECT title,inetref,recordedid FROM recorded WHERE inetref = '' ORDER BY title;
+----------------------------------------+---------+------------+
| title                                  | inetref | recordedid |
+----------------------------------------+---------+------------+
| ABC World News Tonight With David Muir |         |       2766 |
| Chicago Tonight                        |         |       2763 |
| Chicago's Lakefront                    |         |       2667 |
| Close-Up Kings                         |         |       2613 |
| Inside Story: Ferris Bueller's Day Off |         |       2606 |
| NBC5 News 6P                           |         |       2767 |
| Noticias 66 Univisión Chicago          |         |       2764 |
| Noticiero Univisión                    |         |       2765 |
| Political Debate                       |         |       2712 |
| Presidential Address                   |         |        923 |
| Robin Hood                             |         |       2673 |
| The 57th Annual Grammy Awards          |         |       2278 |
| The Oscars                             |         |       2442 |
| WGN Midday News                        |         |       2339 |
+----------------------------------------+---------+------------+
```

**Comment by dmfrey** [comment]
Ok, this breaks what I sis to show the same shows with different inetref's
I did a few weeks ago. Gonna have to rethink this.

**Comment by billmeek** [comment]
When I refresh the Watch Recordings screen, this is used:
http://BE:6544/Dvr/GetTitleInfoList and looking at the
results on my browser, only titles with inetrefs are returned.
So this is working as written in MythTV. And it sounds like
a bug in their query.

What I don't understand is why the episodes appear in
All Recordings.

Looking at the MythTV source GetTitleInfoList excludes 
empty inetrefs:
``` c++
Dvr::GetTitleInfoList()
...
        "SELECT title, inetref, count(title) as count "
        "    FROM recorded "
        "    WHERE inetref <> '' " <-------------------------
        "    AND deletepending = 0 "
        "    GROUP BY title, inetref "
        "    ORDER BY title");
```
And this doesn't exclude empty inetrefs (nor does it return them):
``` C++
 Dvr::GetTitleList(const QString& RecGroup)
"SELECT DISTINCT title FROM recorded "
      "WHERE deletepending = 0";
     " AND recgroup != 'Deleted'";
     " ORDER BY title";
```

**Comment by dmfrey** [comment]
Can you post the json returned from the web service?

**Comment by billmeek** [comment]
``` json
{"TitleInfoList": {"TitleInfos": [{"Title": "Ask This Old House", "Inetref": "ttvdb.py_74326", "Count": "2"},
{"Title": "Battle Creek", "Inetref": "ttvdb.py_274802", "Count": "1"},
{"Title": "Blue Bloods", "Inetref": "ttvdb.py_164981", "Count": "4"},
{"Title": "Castle", "Inetref": "ttvdb.py_83462", "Count": "5"},
{"Title": "Chicago Fire", "Inetref": "ttvdb.py_258541", "Count": "3"},
{"Title": "Chicago PD", "Inetref": "ttvdb.py_269641", "Count": "4"},
{"Title": "Criminal Minds", "Inetref": "ttvdb.py_75710", "Count": "3"},
{"Title": "CSI: Crime Scene Investigation", "Inetref": "ttvdb.py_72546", "Count": "2"},
{"Title": "CSI: Cyber", "Inetref": "ttvdb.py_281551", "Count": "4"},
{"Title": "Elementary", "Inetref": "ttvdb.py_255316", "Count": "3"},
{"Title": "Father Brown", "Inetref": "ttvdb.py_70416", "Count": "1"},
{"Title": "Great Performances", "Inetref": "ttvdb.py_76548", "Count": "2"},
{"Title": "Handel's Messiah: A United States Naval Academy Tradition", "Inetref": "ttvdb.py_242751", "Count": "1"},
{"Title": "Hawaii Five-0", "Inetref": "ttvdb.py_164541", "Count": "4"},
{"Title": "Hometime", "Inetref": "ttvdb.py_82287", "Count": "10"},
{"Title": "Madam Secretary", "Inetref": "ttvdb.py_281623", "Count": "5"},
{"Title": "Married ... With Children", "Inetref": "ttvdb.py_76385", "Count": "1"},
{"Title": "Marvel's Agent Carter", "Inetref": "ttvdb.py_281485", "Count": "1"},
{"Title": "Masterpiece Classic", "Inetref": "ttvdb.py_193131", "Count": "9"},
{"Title": "Masterpiece Classic", "Inetref": "ttvdb.py_261633", "Count": "9"},
{"Title": "Masterpiece Classic", "Inetref": "ttvdb.py_262585", "Count": "1"},
{"Title": "Masterpiece Mystery!", "Inetref": "ttvdb.py_286058", "Count": "1"},
{"Title": "Meet the Press", "Inetref": "ttvdb.py_74657", "Count": "8"},
{"Title": "Men in Black 3", "Inetref": "tmdb3.py_41154", "Count": "1"},
{"Title": "NCIS", "Inetref": "ttvdb.py_72108", "Count": "5"},
{"Title": "NCIS: Los Angeles", "Inetref": "ttvdb.py_95441", "Count": "5"},
{"Title": "NCIS: New Orleans", "Inetref": "ttvdb.py_278125", "Count": "3"},
{"Title": "PBS NewsHour Weekend", "Inetref": "ttvdb.py_273048", "Count": "1"},
{"Title": "Person of Interest", "Inetref": "ttvdb.py_248742", "Count": "6"},
{"Title": "Rehab Addict", "Inetref": "ttvdb.py_267829", "Count": "1"},
{"Title": "Saturday Night Live", "Inetref": "ttvdb.py_76177", "Count": "11"},
{"Title": "Scandal", "Inetref": "ttvdb.py_248841", "Count": "7"},
{"Title": "Scorpion", "Inetref": "ttvdb.py_281630", "Count": "4"},
{"Title": "State of the Union 2015", "Inetref": "ttvdb.py_266860", "Count": "1"},
{"Title": "Swamp Volcano", "Inetref": "tmdb3.py_86971", "Count": "1"},
{"Title": "The Colbert Report", "Inetref": "ttvdb.py_79274", "Count": "3"},
{"Title": "The Daily Show With Jon Stewart", "Inetref": "ttvdb.py_71256", "Count": "12"},
{"Title": "The Dr. Oz Show", "Inetref": "ttvdb.py_123401", "Count": "13"},
{"Title": "The Good Wife", "Inetref": "ttvdb.py_95451", "Count": "5"},
{"Title": "The Mentalist", "Inetref": "ttvdb.py_82459", "Count": "4"},
{"Title": "The Mysteries of Laura", "Inetref": "ttvdb.py_278462", "Count": "4"},
{"Title": "The Replacements", "Inetref": "tmdb3.py_10393", "Count": "1"},
{"Title": "The Roosevelts: An Intimate History", "Inetref": "ttvdb.py_282249", "Count": "7"},
{"Title": "The Royals", "Inetref": "ttvdb.py_279854", "Count": "3"},
{"Title": "This Old House", "Inetref": "ttvdb.py_77444", "Count": "7"},
{"Title": "Transporter: The Series", "Inetref": "ttvdb.py_248962", "Count": "2"},
{"Title": "Undercover Boss", "Inetref": "ttvdb.py_100061", "Count": "2"},
{"Title": "Wizard Wars", "Inetref": "ttvdb.py_283530", "Count": "3"},
{"Title": "Woodsmith Shop", "Inetref": "ttvdb.py_143421", "Count": "2"},
{"Title": "Zero Dark Thirty", "Inetref": "tmdb3.py_97630", "Count": "1"}]}}
```

**Comment by dmfrey** [comment]
I don't see the program in the original question in the TitleInfoList at all in the json. So it shouldn't even be displaying on the Shows screen.  However, if the /Dvr/GetRecordedList returns it with the rest of the programs, it will get entered in the db and be available for view in all recordings.

The Shows screen links either all recordings or an individual recording via the inetref.  The inetref is passed between the activities, or its null to indicate return all recordings from the db.

Does this help?

**Comment by billmeek** [comment]
Yes, GetTitleInfoList is developed by the BE with "WHERE inetref <> '' ",
which explains why the shows aren't there. That's the upstream issue.

I didn't spot the GetRecordedList used for All Recordings, which
explains why the shows are there.

The question becomes what should the app do. Wait for a fix from
MythTV, or use the 'much less efficient' GetRecordedList endpoint
for the Watch Recordings screen?

**Comment by dmfrey** [comment]
GetRecordedList and GetTitleInfoList should run, roughly, every 5 minutes
when connected

**Comment by billmeek** [comment]
The original intent (I think I'm guilty of suggesting the use of
GetTitleInfoList) was to reduce the data passed between the
BE and the app. The idea was to use GetRecordedList with
the TitleRegEx parameter ONLY when displaying the episode
list. TitleRegEx is available in 0.27 and 0.28-pre. 

It sounds like GetTitleInfoList isn't required because all of the
data is now in the app's DB because GetRecordedList runs
(and makes the new search feature work -- so that's very
good.)

Bottom line, there will be users (other than me) that have
titles without inetrefs. Do you want to grab the inetref out 
of the app DB or should I write a "feature request with patch"
in the MythTV Trac system?

**Comment by dmfrey** [comment]
I think for the app, the less data we need to main, the better, so maybe we
need to just user the product table to populate that list.  Maybe mark this
as an enhancement.  Then I can move to the local rowid, instead of
requiring a piece of data from the backend.  It is a bit of a change (like
when i switched from doing it by name), but it is doable.
